### PR TITLE
fccanalysisrun: add --nevents option to limit number of events processed

### PR DESCRIPTION
**Pull request by vvolkl**

### CODE REVIEW

File: config/FCCAnalysisRun.py [hunk 1]
```diff
@@ -292,13 +292,21 @@ def runPreprocess(df):
...
+
+    # cannot use MT with Range()
+    if args.nevents < 0:
+      ROOT.ROOT.EnableImplicitMT(getElement(rdfModule, "nCPUS"))
     ROOT.EnableThreadSafety()
     df = ROOT.RDataFrame("events", inputlist)
 
+    # limit number of events processed
+    if args.nevents > 0:
+      df = df.Range(0, args.nevents)
```
Comment (clementhelsens) [inline config/FCCAnalysisRun.py:304]: should be 
```suggestion
      df = df.Range(0, nevt)
```

File: config/FCCAnalysisRun.py [hunk 2]
```diff
@@ -292,13 +292,21 @@ def runPreprocess(df):
...
-    ROOT.ROOT.EnableImplicitMT(getElement(rdfModule, "nCPUS"))
+
+    # cannot use MT with Range()
+    if args.nevents < 0:
+      ROOT.ROOT.EnableImplicitMT(getElement(rdfModule, "nCPUS"))
     ROOT.EnableThreadSafety()
     df = ROOT.RDataFrame("events", inputlist)
 
+    # limit number of events processed
+    if args.nevents > 0:
```
Comment (clementhelsens) [inline config/FCCAnalysisRun.py:303]: should be
```suggestion
    if nevt > 0:
```

File: config/FCCAnalysisRun.py (hunk 1 above)
Comment (vvolkl) [reply config/FCCAnalysisRun.py:304]: This is the same thing, right? nevt is already set to args.nevents here: https://github.com/HEP-FCC/FCCAnalyses/pull/163/files#diff-0052fde721d1d38ea38d53381326f36e0278d85044d09541927941e18ea0b1feR441

Comment (clementhelsens) [reply config/FCCAnalysisRun.py:304]: ```args``` is not part of ```runRDF``` as it's the args from the parser, not sure this will work

Comment (vvolkl) [reply config/FCCAnalysisRun.py:304]: it works (see checks), `args` is still defined in this scope. And `nevt` will also be positive when `--nevents` is not used, so it adds the call to `Range` even when running over all files, which leads to problem with `ImplicitMT` ...

**Comment by clementhelsens** [comment]
right, I see you define it with negative default value in the parser, looks good

**Pull request merged**